src/context/notification/index.android.js

```diff
@@ -0,0 +1,149 @@
+  permissions: INITIAL_PERMISSIONS,
+  deviceId: getUniqueId(),
+  deviceToken: '',
+};
+
+const initialContextValue = {
+  state: initialState,
+  checkPermissions: () => Promise.resolve(INITIAL_PERMISSIONS),
+  request: () => Promise.resolve(),
+};
+export const NotificationContext = createContext(initialContextValue);
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case UPDATE_PERMISSION:
+      return {
+        ...state,
+        permissions: action.payload,
+      };
+    case UPDATE_DEVICE_TOKEN:
+      return {
+        ...state,
+        deviceToken: action.payload,
+      };
+    default:
+      break;
+  }
+};
+
+export const NotificationProvider = ({children}) => {
+  const [state, dispatch] = useReducer(reducer, initialState);
+  const {authToken} = useContext(AuthContext);
+  const [registerDevice] = useMutationWithAuth(REGISTER_DEVICE);
+  const {data} = useQueryWithAuth(GET_USER_ID, {skip: !authToken});
+  const userId = data?.userProfile?.id;
+
+  useEffect(() => {
+    MailtimePush.init('rewardme', 'RewardMe');
+  }, []);
+
+  const checkPermissions = useCallback(async () => {
+    const deniedPermission = {
+      ...initialState.permissions,
+      alert: false,
+      authorizationStatus: notificationStatus.Denied,
+    };
+    try {
+      const permissionStatus = await RNAndroidNotificationListener.getPermissionStatus();
+      // permissions result can be 'authorized' or 'denied'
+      if (permissionStatus === 'denied') {
+        dispatch({
+          type: UPDATE_PERMISSION,
+          payload: deniedPermission,
+        });
+        return deniedPermission;
+      }
+
+      if (!state.deviceToken) {
+        // android: update device token once permissions is authorized
+        const mtPush = await MailtimePush.register(userId);
+        dispatch({type: UPDATE_DEVICE_TOKEN, payload: mtPush.token});
+      }
+      return initialState.permissions;
+    } catch (e) {
+      console.error('error check permission android', e);
+      dispatch({type: UPDATE_PERMISSION, payload: deniedPermission});
+      return deniedPermission;
+    }
+  }, [userId, state.deviceToken]);
+
+  const request = useCallback(async () => {
+    try {
+      return await RNAndroidNotificationListener.requestPermission();
+    } catch (e) {
+      console.error('error request permission android');
+      return {};
+    }
+  }, []);
+
+  const notificationContext = useMemo(
+    () => ({
+      checkPermissions,
+      request,
+      state,
+    }),
+    [checkPermissions, request, state],
+  );
+
+  useEffect(() => {
+    checkPermissions();
+  }, [checkPermissions]);
+
+  useEffect(() => {
+    const _registerDevice = async () => {
+      try {
+        const mtPush = await MailtimePush.register(userId);
+        await registerDevice({
+          variables: {
+            deviceId: state.deviceId,
+            platform: mtPush.platform,
+            pushToken: mtPush.token,
+          },
+        });
```

Comment:
Do not need to register again. Use saved token and platform from state:
```suggestion
        await registerDevice({
          variables: {
            deviceId: state.deviceId,
            platform: state.platform,
            pushToken: state.token,
          },
        });
```